config/externalname.go

```diff
@@ -3009,6 +3020,19 @@ func FormattedIdentifierFromProvider(separator string, keys ...string) config.Ex
 	return e
 }
 
+// FormattedIdentifierFromProviderWithIdentifierFields is a helper function to construct Terraform
+// IDs that use elements from the parameters in a certain string format.
+// It should be used in cases where all information in the ID is gathered from
+// the spec and not user defined like name. For example, zone_id:vpc_id.
+// This function sets the keys as IdentifierFields, which means that they are always required, even for observe-only
+// resources. Because the id is constructed exclusively from the keys, omitting them (even if the external name
+// annotation is set) leaves the provider unable to find the terraform id to use to observe the resource.
+func FormattedIdentifierFromProviderWithIdentifierFields(separator string, keys ...string) config.ExternalName {
```

Comment:
These function names are getting excessively verbose, but I couldn't think of another option that preserved the "dsl-like" semantics of (most of) the external name config file, didn't introduce a schema change, and could handle the `...string` variable arguments. 

If anyone has a better idea I'd love to hear it.

Reply:
When I examined the import sections in the two resources you added, the parameters used in id can be taken from `spec.forProvider`. In other words, since these are not resources whose identities are assigned by the remote client, I think it is better to use [TemplatedStringAsIdentifier](https://github.com/crossplane/upjet/blob/37c7f4e91d57982f58e496dfb86dd7c2d29a1921/pkg/config/externalname.go#L70) here if it works.

This function also sets the keys as `IdentifierFields`, so extra configuration is unnecessary.

Reply:
That's exactly what I tried as my first external name configuration; see https://github.com/crossplane-contrib/provider-upjet-aws/pull/1273/commits/24a791a60d863aeebc1b88c91b15540fc564d514. The problem with `TemplatedStringAsIdentifier` is that because it has `DisableNameInitializer: false` it initializes the external name to the `metadata.name`, before updating it to use the value from the template. When I was testing, this caused something to not work; I don't remember what it was. 

Your comment about the identifier not really coming from the provider is a good one; that means this is not a good name for this function. Really the essence of what I need is to disable the name initializer. I think the best solution is to add `WithDisabledNameInitializer` to the `ExternalNameFrom` helper functions @ulucinar added in https://github.com/crossplane/upjet/pull/342/commits/47cb5c5b4bba12627007155937ca23bfc468c0ba

Reply:
The trouble is that I couldn't find any actual examples of usages of the `ExternalNameFrom` helpers, so I'll probably have to fumble around a bit in order to figure out the syntax.

Reply:
If the problem is just `DisableNameInitializer: false`, we may use [this function](https://github.com/crossplane-contrib/provider-upjet-aws/blob/be19a8afe756d43e5a79105766001bbb1ca300bd/config/externalname.go#L3095)?

Reply:
`TemplatedStringAsIdentifierWithNoName` is only used by two resources, both of which have a provider-generated component to their terraform id which is separated out and stored in the external name. I recall running into some problems with them in the past related to initial creation, which prompted the TODO I put in that function's docs, and I'm re-running their uptest examples to refresh my memory. I'll check back once those tests finish to see what they did.

Do you have a reason for preferring the "Templated" syntax in the external name config over specifying the parameters and the separator? It seems like that just creates more work to reverse-engineer the template string to do things like find identifier fields and look for `{{ .external_name }}`.

Reply:
I think that whatever problem I was observing on initial creation for `TemplatedStringAsIdentifierWithNoName` was only a problem with the cli-forking architecture, as I saw no issues with those resources. I'll remove the TODO comment from that external name config.

Reply:
> Do you have a reason for preferring the "Templated" syntax in the external name config over specifying the parameters and the separator? 

I don't have any special reason or desire to use the templated syntax. I don't find it appropriate to add a new function for consistency reasons if we can configure the resource without any problems with the existing external name functions/configurations.

> I think that whatever problem I was observing on initial creation for TemplatedStringAsIdentifierWithNoName was only a problem with the cli-forking architecture, as I saw no issues with those resources. I'll remove the TODO comment from that external name config.

From your comment, am I to understand that you will try to continue with `TemplatedStringAsIdentifierWithNoName`?